Source/Server/equellaserver/build.sbt

```
@@ -116,7 +116,9 @@ libraryDependencies ++= Seq(
   "javax.xml"                 % "jaxrpc"                   % "1.1",
   "jdom"                      % "jdom"                     % "1.1",
   "com.github.equella.jpf"    % "jpf"                      % "1.0.7",
-  "log4j"                     % "log4j"                    % "1.2.17",
+  "org.apache.logging.log4j"  % "log4j"                    % "2.17.1",
+  "org.apache.logging.log4j"  % "log4j-slf4j-impl"         % "2.17.1",
+  "org.apache.logging.log4j"  % "log4j-1.2-api"            % "2.17.1",
```

Hmm, shouldn't this just be:

```suggestion
  "org.apache.logging.log4j"  % "log4j-api"            % "2.17.1",
```

And, do we need the API dep? Kind of feels like it should be a transitive for log4j-slf4j-impl. :thinking:

---

Actually, `log4j-1.2-api` is a bridge which allows log4j2 to be used with some old log4j code. 

For example, in `ItemServiceImpl` there is usage of `MDC` which is replaced by `Thread Context Map`. And in 
`AuthenticatedThread` there is usage of `NDC` which is replaced by `Thread Context Stack`. 

I did try to update the code to use the new stuff, but I stopped and then forgot...I will try again.